File: code-example-tests/csharp/driver/Examples/Aggregation/Builders/ReplaceWith.cs
using DotNetEnv;
using MongoDB.Driver;

namespace Examples.Aggregation.Builders;

public class ReplaceWithExample
{
    private readonly string _uri = Env.GetString("CONNECTION_STRING",
        "Set your CONNECTION_STRING in the .env file");

    public List<ImdbData?> RunReplaceWithPipeline()
    {
        // :snippet-start: replace-with
        var pipeline = new EmptyPipelineDefinition<Movie>()
            .Sort(Builders<Movie>.Sort.Ascending(m => m.Title))
            .Limit(5)
            .ReplaceWith(m => m.Imdb);
        // :snippet-end:
        using var client = new MongoClient(_uri);
        var collection = client
            .GetDatabase("sample_mflix")
            .GetCollection<Movie>("movies");
        return collection.Aggregate(pipeline).ToList();
    }
}

File: code-example-tests/javascript/driver/examples/aggregation/stages/replace-with/replace-with.js
//	:replace-start: {
//	  "terms":
//	  {
//	  "process.env.CONNECTION_STRING": "\"<connection string URI>\"",
//    "export ": ""
//	  }
//	}
import { MongoClient } from 'mongodb';

export async function runReplaceWith() {
  const uri = process.env.CONNECTION_STRING;
  const client = new MongoClient(uri);

  try {
    const database = client.db('sample_mflix');
    const collection = database.collection('movies');

    // :snippet-start: replace-with
    const pipeline = [{ $replaceWith: '$imdb' }];

    const cursor = collection.aggregate(pipeline);
    // :uncomment-start:
    // return cursor;
    // :uncomment-end:
    // :snippet-end:

    const documents = [];
    for await (const document of cursor) {
      documents.push(document);
      if (documents.length >= 5) break;
    }
    return documents;
  } finally {
    await client.close();
  }
}
// :uncomment-start:
//runReplaceWith().catch(console.dir);
// :uncomment-end:
// :replace-end:
